src/Apps/Settings/Routes/MyCollection/Components/MyCollectionArtworkGrid.tsx

```diff
@@ -0,0 +1,249 @@
+import { AuthContextModule } from "@artsy/cohesion"
+import { Flex } from "@artsy/palette"
+import { ArtworkGridItemFragmentContainer } from "Components/Artwork/GridItem"
+import {
+  areSectionedArtworksEqual,
+  createSectionedArtworks,
+} from "Components/ArtworkGrid"
+import { ArtworkGridEmptyState } from "Components/ArtworkGrid/ArtworkGridEmptyState"
+import { isEmpty, isEqual } from "lodash"
+import memoizeOnce from "memoize-one"
+import * as React from "react"
+import ReactDOM from "react-dom"
+import { createFragmentContainer, graphql } from "react-relay"
+import styled from "styled-components"
+import { StoredImage } from "Utils/localImagesHelpers"
+import { Media, valuesWithBreakpointProps } from "Utils/Responsive"
+import { MyCollectionArtworkGrid_artworks$data } from "__generated__/MyCollectionArtworkGrid_artworks.graphql"
+
+// @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
+type Artwork = MyCollectionArtworkGrid_artworks$data["edges"][0]["node"]
+
+type SectionedArtworks = Array<Array<Artwork>>
+
+export interface MyCollectionArtworkGridProps
+  extends React.HTMLProps<HTMLDivElement> {
+  artworks: MyCollectionArtworkGrid_artworks$data
+  contextModule?: AuthContextModule
+  columnCount: number | number[]
+  getLocalImageSrcByArtworkID: (artworkID: string) => StoredImage | null
+  isCollectorProfileEnabled?: boolean | null
+  preloadImageCount?: number
+  itemMargin?: number
+  onLoadMore?: () => any
+  sectionMargin?: number
+}
+
+export interface ArtworkGridContainerState {
+  loading: boolean
+  interval: any
+}
+
+export class MyCollectionArtworkGridContainer extends React.Component<
```

Comment:
Adding the interface `ArtworkConnectionInterface` to the connection doesn't work either because Relay complains that the connection cannot be both `ArtworkConnectionInterface` and `MyCollectionConnection`. I'm not sure if there is a way that would solve this issue on a GraphQL level.

Reply:
cc @mzikherman — is there a way around this?

Its looking like I _really_ need to reimplement bin packing with a generic/non-fragment interface.

Reply:
(If there's no way around this for now I would add a big comment explaining this)

Reply:
[I just found a way to avoid the code duplication by just creating a new fragment container (using `ArtworkGrid`).](https://github.com/artsy/force/pull/11705/commits/a3ab627eda9f05a38edf88b26e0e1f8fbf5b6947)